server/src/routes/tasksRoute.ts

```diff
@@ -1,12 +1,18 @@
 import express from 'express';
-import { checkAuth } from '../middlewares';
-import { addTask, getTasks, getTasksSection,editTask, deleteTask } from '../controllers';
+import { asyncMiddleware, checkAuth } from '../middlewares';
+import {
+  addTask,
+  getTasks,
+  getTasksSection,
+  editTask,
+  deleteTask,
+} from '../controllers';
 
 const tasksRoute = express.Router();
-tasksRoute.post('/addTask', addTask);
-tasksRoute.get('/tasks', checkAuth, getTasks);
-tasksRoute.get('/tasks/:id', getTasksSection);
-tasksRoute.put('/task/:id', checkAuth, editTask);
-tasksRoute.delete('/deleteTask/:id', checkAuth, deleteTask);
-
+tasksRoute.use(checkAuth);
```

Comment:
check auth also has try and catch, so we can wrap our middleware with async-middleware and remove try-catch block